src/basic/compress.c

```diff
@@ -109,10 +109,10 @@ static inline void lzma_end_wrapper(lzma_stream *ls) {
 #define ALIGN_8(l) ALIGN_TO(l, sizeof(size_t))
 
 static const char* const compression_table[_COMPRESSION_MAX] = {
-        [COMPRESSION_NONE] = "NONE",
```

Comment:
this breaks backward compatibility, so at the very least the parser needs to accept both lower and upper case entries

Reply:
https://github.com/systemd/systemd/pull/34595#discussion_r1806329475

Reply:
It is only used in the environment variable, so I did not think it is necessary to keep backward compat.
But, ok, let's take a safer way. 
Please keep the env var to take the upper case.
The simplest way is also introduce the string table of upper case.